src/wrapper/qr_mumps_api.jl

```diff
@@ -3,47 +3,35 @@ for (fname, lname, elty, subty) in (("sqrm_spmat_init_c", libsqrm, Float32   , F
                                     ("cqrm_spmat_init_c", libcqrm, ComplexF32, Float32),
                                     ("zqrm_spmat_init_c", libzqrm, ComplexF64, Float64))
     @eval begin
-        function qrm_spmat_init!(spmat :: qrm_spmat{$elty})
-            err = ccall(($fname, $lname), Cint, (Ref{qrm_spmat{$elty}},), spmat)
+        function qrm_spmat_init!(spmat :: c_spmat{$elty})
+            err = ccall(($fname, $lname), Cint, (Ref{c_spmat{$elty}},), spmat)
             (err ≠ 0) && throw(ErrorException(error_handling(err)))
             return nothing
         end
 
         function qrm_spmat_init(::Type{$elty})
-            spmat = qrm_spmat{$elty}()
-            qrm_spmat_init!(spmat)
+            spmat2 = c_spmat{$elty}()
+            qrm_spmat_init!(spmat2)
+            spmat = qrm_spmat{$elty}(spmat2.h)
             return spmat
         end
 
-        function qrm_spmat_init!(spmat :: qrm_spmat{$elty}, A :: SparseMatrixCSC{$elty,I}; sym :: Bool=false) where I <: Integer
-            err = ccall(($fname, $lname), Cint, (Ref{qrm_spmat{$elty}},), spmat)
-            (err ≠ 0) && throw(ErrorException(error_handling(err)))
+        function qrm_spmat_init(A :: SparseMatrixCSC{$elty,I}; sym :: Bool=false) where I <: Integer
+            spmat = qrm_spmat_init($elty)
             irn, jcn, val = findnz(A)
             m, n = size(A)
             nz = nnz(A)
-            irn = Int32.(irn)
-            jcn = Int32.(jcn)
-            val = $elty.(val)
-            spmat.irn = pointer(irn)
-            spmat.jcn = pointer(jcn)
-            spmat.val = pointer(val)
+            spmat.irn = Int32.(irn)
+            spmat.jcn = Int32.(jcn)
+            spmat.val = $elty.(val)
             spmat.m   = Int32(m)
             spmat.n   = Int32(n)
             spmat.nz  = Int32(nz)
             spmat.sym = Int32(sym)
```

Comment:
also on assignment to a struct an appropriate convert is invoked, so no need to be explicit here;
(explicit converts for vectors were needed before, since `pointers` to them were recorded)

Reply:
Thanks !